### PR TITLE
Add comma-dangle for functions Fixes #951

### LINKED ISSUE #951: Trailing commas in function arguments
What about trailing commas here?

``` javascript
const hero = (
  firstName,
  lastName,
  birthYear,
  superPower, // <-- this one
) => {
  // ....
};
```

It needs [http://babeljs.io/docs/plugins/syntax-trailing-function-commas/](http://babeljs.io/docs/plugins/syntax-trailing-function-commas/) and it's based on [this proposal](https://github.com/jeffmo/es-trailing-function-commas)

Does AirBnB encourages the use of trailing commas here?

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: packages/eslint-config-airbnb-base/rules/errors.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 module.exports = {
   rules: {
     // require trailing commas in multiline object literals
-    'comma-dangle': ['error', 'always-multiline'],
+    'comma-dangle': ['error', {
+      objects: 'always-multiline',
```
Comment (ljharb) [inline packages/eslint-config-airbnb-base/rules/errors.js]: this should also include `arrays` set to "always-multiline"

**Comment by ljharb** [comment]
Also, we'll need to update https://github.com/airbnb/javascript#commas--dangling to have examples of both multiline function signatures and invocations

File: README.md [hunk 2]
```diff
@@ -2397,6 +2397,45 @@ Other Style Guides
...
+      firstName,
+      lastName,
+      inventorOf,
+    ) {
+      // does nothing
+    }
+    
+    // good
+    function createHero(firstName,
+                        lastName,
```
Comment (ljharb) [inline README.md]: this is actually a bad example.

Comment (graingert) [inline README.md]: It's not, because the rule is specifically designed to allow it.

Comment (ljharb) [inline README.md]: It's a bad example for our styleguide - that style is absolutely not permitted, even if `comma-dangle` works with it. Other linter rules should be forbidding it.

Comment (ljharb) [inline README.md]: Specifically, "good" doesn't mean "passes the linter rule", "good" means "conforms to our styleguide"

**Comment by graingert** [comment]
Where is this style banned?

**Comment by ljharb** [comment]
It seems we don't have a specific section on it in https://github.com/airbnb/javascript#functions - I'll add one now.

In general, in any case, any time the entire `( )`, `[ ]`, or `{ }` doesn't fit on a single line, the `(`/`[`/`{` must end a line, the `)`/`]`/`}` must begin a line, and every item must be on a line by itself.

**Comment by graingert** [comment]
You should totally request a rule for that from eslint

**Comment by ljharb** [comment]
Section added in c51251d17b8e38f1e8975a5e4eb13d7986229410

File: README.md (hunk 2 above)
Comment (graingert) [inline README.md]: I've marked this branch as editable by maintainers. Otherwise you can wait a few hours for me to get to a keyboard

**Pull request merged**